packages/core/EventManager.ts

```diff
@@ -535,78 +535,105 @@ export default class EventManager {
     };
 
     if (event.destinationCalendar && event.destinationCalendar.length > 0) {
-      // Since GCal pushes events to multiple calendars we only want to create one event per booking
-      let gCalAdded = false;
-      const destinationCalendars: DestinationCalendar[] = event.destinationCalendar.reduce(
```

Comment:
Is the reduce refactor necessary to fix the bug? Or was it done for getting two birds with one stone?

If it's the latter then I'm afraid it is making the PR harder to review since it's not clear "where" is the bug that we're fixing.

Also as @keithwillcode mentioned we don't have EventManager tests to ship this PR with 100% confidence. RN if I approve this I'm not 100% confident this will not break other event type configurations.

My suggestion would be:

Make a PR with "only" the changes needed to fix the current bug. Then other PR for adding tests. Then other PR to do chores and refactoring.

Reply:
The refactor is necessary since ideally only one calendar event is created for the booking. When inviting team members it should add events to their default destination calendars by inviting those calendars to the event.

Since this is affecting collective events, a PR with "only" the fix would be to remove pushing multiple destination calendars when the event is a collective one. That would be the safest bet. Looking through the `CalendarService` files of GCal and Outlook we already add team members as the attendees. Although for Outlook we still need to process the `externalId` properly so all team members are added to the event.

Reply:
Yeah I think we should focus the rest of day on getting tests wired up here and we ship this tomorrow. 

This is the type of change that is high risk of making things worse than the original bug we are trying to fix.

Reply:
The tests should written against main and not against this code first to verify all existing logic stays the same. Once verified, we then add tests to verify this change in logic